### PR TITLE
Improve monorepo setup and tooling

### PR DESCRIPTION
**Pull request by emmenko**
* use node v14
* use preconstruct for linking and bundling packages
* use manypkg
* move monorepo tools to main root folder, including jest, tsconfig, etc.
* use latest husky
* fix ts errors
* remove lerna
* add contributing document

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -15,59 +15,58 @@ aliases:
       - .cache
       - node_modules
 
+executors:
+  node_14:
+    docker:
+      - image: cimg/node:14.17.4
```
Comment (emmenko) [inline .circleci/config.yml:21]: Optimized image for circleci

File: README.md [hunk 2]
```diff
@@ -23,10 +20,13 @@ If you have any urgent issues regarding this repository please create a support
 | [`platform-sdk`](/packages/platform-sdk)   | [![platform-sdk Version][platform-sdk-icon]][platform-sdk-version]    |
 | [`importapi-sdk`](/packages/importapi-sdk) | [![importapi-sdk Version][importapi-sdk-icon]][importapi-sdk-version] |
 | [`ml-sdk`](/packages/ml-sdk)               | [![ml-sdk Version][ml-sdk-icon]][ml-sdk-version]                      |
+| [`history-sdk`](/packages/history-sdk)     | [![history-sdk Version][history-sdk-icon]][history-sdk-version]       |
```
Comment (emmenko) [inline README.md:23]: Was it forgotten or was it omitted on purpose?

File: jest.config.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  testRegex: '\\.(test|spec)\\.[j|t]sx?$',
+  moduleFileExtensions: ['ts', 'js', 'json'],
+  watchPlugins: ['jest-watch-typeahead/filename'],
+  reporters: [
+    'default',
+    process.env.CI === 'true'
+      ? [
+          'jest-junit',
+          { outputName: 'results.xml', outputDirectory: 'test-results' },
+        ]
```
Comment (emmenko) [inline jest.config.js:20]: Do you know why are we using and storing test results as artifacts? Are they used somewhere?

File: packages/history-sdk/test/helpers/api-helpers.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import { createClient } from '@commercetools/sdk-client'
```
Comment (emmenko) [inline packages/history-sdk/test/helpers/api-helpers.ts:1]: This file was missing, so I more or less copied the setup from the ML package.

File: packages/history-sdk/test/helpers/ctp-api-helper.ts [hunk 5]
```diff
@@ -5,15 +5,15 @@ import {
   ApiRoot,
   createExecutorFromMiddlewares,
   executeRequest,
-} from './../../src'
+} from '@commercetools/platform-sdk'
```
Comment (emmenko) [inline packages/history-sdk/test/helpers/ctp-api-helper.ts:8]: This was wrong (according to how the other packages have the same setup).

File: README.md (hunk 2 above)
Comment (ajimae) [reply README.md:23]: Note sure, but I think it's fine to have it as part of the packages listed in the README.md

File: jest.config.js (hunk 3 above)
Comment (ajimae) [reply jest.config.js:20]: Not really, but I will also look into it.

**Pull request merged**